apps/passport-client/src/validateState.ts

```diff
@@ -196,6 +254,20 @@ export function getRunningAppStateValidationErrors(
     // places that the user's commitment exists match - in the self, the
     // identity, and in the pcd collection
 
+    if (publicKeyV4FromSelfField !== publicKeyV4FromIdentityField) {
```

Comment:
Why are these two clauses here rather than in the corresponding else clause for V4 below (line 299)?  The public key check here seems like it belongs there.  The commitment check here seems to duplicate what's done below.

Reply:
To be extra careful, you could check the public key separate from the commitment, and also confirm they relate to each other properly:
- If commitment is present/abssent, pubkey is also present/absent
- Check that commitment correctly reflects what would be derived from the public key